src/services/AuroraDbService.js

```diff
@@ -45,6 +47,30 @@ export default class AuroraDbService {
     return resp.records[0];
   }
 
+  async selectPaginate(
+    sql,
+    sqlParams,
+    perPage = 10,
+    currentPage = 1,
+    total = null
+  ) {
+    let paginator;
+    if (typeof total === 'number') {
+      paginator = new LengthAwarePaginator(
+        this,
+        sql,
+        sqlParams,
+        perPage,
+        currentPage,
+        total
```

Comment:
I can now confirm that it is all good as the `=== true` is actually not necessary.

Reply:
They are actually important if we want to use LengthAwarePaginator on the auroradb with it without supplying the total number and use the internally available logic to get the total count of the query.

In the current code right now, the LengthAwarePaginator will only be used if the total params is supplied by a number, which will represent the total count, which can be derived somewhere, say, somebody has created a more efficient query to get the total number.

If we want the AuroraDbService to use LengthAwarePaginator  and  make it calculate the total count of the query automatically, we have to supply the boolean true , then the condition inside will determine if LengthAwarePaginator should be supplied with a number or null.

Reply:
My apologies for the late realization.

Reply:
If the LengthAwarePaginator requires it to be a number, than it should do the check within the LengthAwarePaginator and not on the function that is calling it. This is to prevent other functions from misusing it as well. Which I believe is already handled and will throw an Error if it is not null nor a number.

If this function requires a total to be a null or a number, than it needs the same validation as well. But this is not the explicit requirement of this selectPaginate function correct?

Reply:
Yes, that is not the explicit requirement of the `selectPaginate` function. The additional condition is specific for the `selectPaginate` function because that determines whether to create a `Paginator` or a `LengthAwarePaginator`.